Introduction to state: Update 3rd assignment step with a guide for how to fork the codesandbox (#28309)
TheOdinProject/curriculum: react/states_and_effects/introduction_to_state.md

    1. [State: A Component's Memory](https://react.dev/learn/state-a-components-memory)
    1. [Render and Commit](https://react.dev/learn/render-and-commit)
1. Read this [article on React Reconciliation Algorithm](https://medium.com/javarevisited/react-reconciliation-algorithm-86e3e22c1b40) for a great explanation.

> 1. Head back to the colour changing background example from earlier in the lesson, and add a new state variable to keep track of the number of times the background color has been changed. Display the number of times the background color has been changed on the page. To do this you'll need your own account to create a fork. So first, click open editor in the top right of the preview to head to the codesandbox, then if you don't already have an account go ahead and make one by linking your GitHub account. This will allow you create your own fork, which will be an option in the top right.

Reviewer: Not a hard block, but I think we can simplify the text a little. The below is just a suggestion. What do you think?

```suggestion
1. Head back to the colour changing background example from earlier in the lesson, and add a new state variable to keep track of the number of times the background color has been changed. Display the number of times the background color has been changed on the page. You will need to fork the codesandbox to do this, which requires a codesandbox account. Click "Open Editor" in the top right of the codesandbox embed, then "Fork" in the top right of the editor.
```

Author: I like the changes you've suggested. I think the double quotes helps make the actions stand out a lot! I was trying to achieve something similar and I'm gonna use that idea in the future.

Author: Is there anything I need to do on my side so this is added to the PR?

Reviewer: Commit and push the change.

You can make the change locally then commit and push to the branch. Alternatively, you can hit the "commit suggestion" button if you like the suggested change exactly as it is, to commit that exact change to this branch. For future reference, note that doing this will put your local branch out of sync with this remote branch, so you'd have to pull from remote to update your local should you want to make any further changes locally. Otherwise you'd end up with weird git timelines conflicting and headaches if you aren't familiar.

Author: That makes sense. I'll keep it in mind for the future, thanks for the tip!

Author: I noticed a tick box, on the right-hand side, when making pull requests. It reads '**Allow edits and access to secrets by maintainers**' Is this useful for me to check, for when you or other maintainers need to make improvements to my pull requests?

Reviewer: You can tick it, yes. Generally, we prefer requesting changes from the PR author. Since we're all volunteers, that includes contributors and it's not that often that we'd have a change that's so urgent to merge that we absolutely can't wait for the author to implement the request. Doesn't hurt to have it ticked, we just seldom need to be able to do so here.
